### ISSUE TITLE
What has appended to you?

### ISSUE THREAD

**Issue by dhmiller** [mancoast/mb3d#13]
Such potential - is it ongoing?

**Comment by mancoast** [comment]
Greetings! This project is still on my radar, but I have been really busy as of lately. There is future for this software, it needs much work to rid of the bitrot

**Comment by dhmiller** [comment]
Good to hear. Could we incentivize you in anyway? I bet people would pay
for a 64 bit version - I certainly would. And so many bugs!!  The Windows
beta version puts the various Navigator windows on top of each other so you
cannot say, Reset any changes you make to the formula parameter values.
Anyway, please keep me posted and I will  pass the word along to the
forums.. Very thankful for your efforts and wish I could help .
d.

**Comment by coast77777777777** [comment]
@mclarekin do you have any time to take a look at this?

**Comment by coast77777777777** [comment]
@dhmiller It would be ideal to fund our friend @mclarekin

**Comment by dhmiller** [comment]
? Can you give me a little more info about your request? Is this Bob from
the FB page? Any news on your progress?

**Comment by coast77777777777** [comment]
Greetings, yes this is bob. I changed my github name. Currently we are making progress, ideally GPU acceleration is the ultimate goal for MB3D settings.

**Comment by dhmiller** [comment]
Thanks Bob. Great news to know that you were back on the case. I can continue to put up bug issues if you’re interested in those. And who is the guy that you were suggesting get funding?

**Comment by mclarekin** [comment]
I think

a) main focus is that it would be nice to have M3D converted into a 64bit
GPU program

b)  more formulas and transforms could be added , ( a minor thing)

c)  structural changes, to remove limitations  ( a major thing)


Currently those who are more familiar with the M3D program, (Thargor ,
Darkbeam etc) consider future  development  to be a waste of time.

Programs get developed as the programmer sees it at that time. Later there
always appear some structural and backwards compatibility problems that
occur when trying to develop it further.

MandelbulberV2  already requires  some annoying backwards compatibility
code. I would imagine that M3D could be even worse.
Fragmenatrium is kept much simpler  so it does not suffer too much, but
users have to change their code at times to run older parameters

Concentrate on a)  and think about b) & c) later.

**Comment by mclarekin** [comment]
@dhmiller  I work on MandelbulberV2 like Bob
You could crowd fun new programmers to work on the project, but  for people to then buy a copy would involve issues.
BTW we have already  had someone selling free open source Mandelbulber by simply "adding value" by putting it onto a CD (he also claimed that a percentage would go to the developers LOL)

**Comment by dhmiller** [comment]
I agree with the 64-bit priority, but I think some minor bug fixes would
make peoples' lives a lot easier. And I know you must be working on
Windows, but there seem to be a vast number of people who would like a
stable Mac version. High Sierra generally works fine with Wine, but Mojave
is a disaster, and the new O/S will probably kill it completely. Not sure
who could tackle that.

**Comment by mclarekin** [comment]
About  MacOS    i have some memory   that maybe fractalforums   member
WAUthethird built a Mac version

**Comment by dhmiller** [comment]
Thanks for that - can't seem to track him (her?) down and wonder if it is a
real build or just Wined... So I guess you guys don't plan any Mac build..
.But doesn't Mandelbulber have a native Mac version? And is the idea to
build a converter M3D - Mandelbulber... not clear on your plans.
Please keep me up to date on your progress - this is a very big deal for
the M3D community, so thanks again for your efforts.

**Comment by coast77777777777** [comment]
@dhmiller  macOS is top priority along with continuous integration. (Appveyor) 
@mclarekin do you have apple hardware?

**Comment by dhmiller** [comment]
I have a bunch of macs including a Mac Pro,  A nice iMac, and two MacBook Pros. I also have a very nice HP workstation that was top-of-the-line two years back and all the machines are pretty much rendering 24 seven 
www.dennishmiller.com
See computer graphics for what it’s worth

Totally immersed in the software and happy to help in anyway I can including some finances if that’s what it would take.

**Comment by coast77777777777** [comment]
The software can compile for macOS, rad studio is fantastic. However, I have no previous experience with the MB3D system so I cannot tell what is broken in macOS.

**Comment by dhmiller** [comment]
I can help compile bug  reports and features requests if that’s what you’re looking for but I’m not much use tracking down code issues. What would help here?

**Comment by dhmiller** [comment]
A native build of 1.99 for the Mac warts and all would be a huge deal for the community. Do you have Mac hardware to compile it? My brother was a senior guy at IBM for 30 years and I can ask him but I don’t know where the code is or how clean it might be because I don’t think he would want to do a lot of work on it.

**Comment by mclarekin** [comment]
I do not have a Mac

@ dhmiller,  you could look as this
https://fractalforums.org/downloads/10/mandelbulb3dv199-mac-version-premade-app/1047/msg15405#msg15405

**Comment by dhmiller** [comment]
Hey guys. Heres' what my brother wrote when I asked him if he could compile
a native Mac version:
"For the build, if it's an application then it depends on what language
it's written in and how many of the platform APIs it uses.  It's probably
pretty difficult to port without very good knowledge of the APIs available
on each platform.  If it's written in an interpreted language then less
difficult.  But even 1/2-way compiled languages like Python or Java have
small differences platform to platform."


So where exactly is the code that Bob used and would you say it is even
"reasonably" clean at this point? Obviously Bob was able to compile a very
stable version for Windows, so I'm hoping my brother could do a Mac version.
Could you let me know what you think?
Thanks much

**Comment by dhmiller** [comment]
Also, thanks for the link. I have made wine wrapped versions myself that work fine under High Sierra but are a disaster under Mojave. I think I’ve even tried this one, but I’m really looking for a native Mac compiled version because after Mojave, things are going to get even stricter on the Mac. I’d love to have my brother take a look at the code, so if you can tell me where to find the cleanest version, that would be great. Maybe Bob as that?

**Comment by dhmiller** [comment]
Bob - can you tell me where we could find your improved 2.0 alpha code? Id
like to have my brother take a look and see what he thinks about compiling
a native Mac version.
Thanks very much,
Dennis

**Comment by dhmiller** [comment]
Could you tell me what state the most recent code is and where we could get
access to it? Still hoping my brother could compile a native Mac build.
(Hope you're getting these emails...)
Thanks much
Dennis

**Comment by mclarekin** [comment]
From my limited understanding of this project

This is where Bob's files are,  which  come from Thagor6, who was the last
person to implement new things, (maybe 2 years ago).


https://github.com/coast77777777777/mb3d

**Comment by dhmiller** [comment]
Thanks very much. So can you tell me what the nature of the collaboration with Bob in respect to Mandelbulb 3D is? There are dozens of curious people at the FB group...

**Comment by mclarekin** [comment]
I am not sure what Bob's current plans are.

I am not a programmer, but if M3D was converted to a language I could read,
i could possibilly add new formulas. That is the limit of my potential
collaboration with Bob in respect to Mandelbulb 3D

**Comment by coast77777777777** [comment]
I will get this system using appveyor or Travis like we have done with mandelbulber2.  That means each change of code to this repo will result with a new MB3D.exe file. Then implement a few easy things, and do another release for MacOS and Windows 32bit

**Comment by dhmiller** [comment]
That sounds great, Bob. The Mac version is somewhat pressing since Mojave
is already a bad deal for Wine and the next version won't support any 32
bit apps. Imho, the Win version is pretty stable, so just curious what sort
of time frame you could estimate for these builds.
And is the idea to have a converter for Mandelbulber, or was that just
speculation on the FB page...
Thanks again,
Dennis

**Comment by coast77777777777** [comment]
Greetings Dennis,

My plan is to get cloud compilation functionality for this MB3D system. The converter from MB3D to Mandelbulber2 is also happening, that is not speculation, it is fact. It will take time, but it’s something that is worth the wait. 

Thanks,
Rob

**Comment by dhmiller** [comment]
Would cloud compilation facilitate a native Mac build by any chance?

**Comment by dhmiller** [comment]
Hi Bob. Apparently Catalina is the new OS for Mac and coming soon. Do you
have any sense when there might be a new Mac OS build, which would mitigate
needing Wine to run on thew Mac, which I am certain will not work at all,
given the massive problems with Mojave, the current OS? Just curious if a
native Mac build is on the radar.
Thanks much and hope the work is going well.
Dennis
